pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -22,34 +26,53 @@ var systemRegistriesConfPath = builtinRegistriesConfPath
 // DO NOT change this, instead see systemRegistriesConfPath above.
 const builtinRegistriesConfPath = "/etc/containers/registries.conf"
 
-// Mirror represents a mirror. Mirrors can be used as pull-through caches for
-// registries.
-type Mirror struct {
-	// The mirror's URL.
-	URL string `toml:"url"`
+// Endpoint describes a remote location of a registry.
+type Endpoint struct {
+	// The endpoint's remote location.
+	Location string `toml:"location"`
 	// If true, certs verification will be skipped and HTTP (non-TLS)
 	// connections will be allowed.
 	Insecure bool `toml:"insecure"`
 }
 
+// RewriteReference will substitute the provided reference `prefix` to the
+// endpoints `location` from the `ref` and creates a new named reference from it.
+// The function errors if the newly created reference is not parsable.
+func (e *Endpoint) RewriteReference(ref reference.Named, prefix string) (reference.Named, error) {
+	if ref == nil {
+		return nil, fmt.Errorf("provided reference is nil")
+	}
+	if prefix == "" {
+		return ref, nil
+	}
+	refString := ref.String()
+	if refMatchesPrefix(refString, prefix) {
+		newNamedRef := strings.Replace(refString, prefix, e.Location, 1)
+		newParsedRef, err := reference.ParseNamed(newNamedRef)
+		if newParsedRef != nil {
+			logrus.Debugf("reference rewritten from '%v' to '%v'", refString, newParsedRef.String())
+		}
+		return newParsedRef, errors.Wrapf(err, "error rewriting reference")
```

Comment:
Can we rewrite that to:
```
if err != nil {
   return nil, errors.Wrapf(...)
}
return newParsedRef, nil
```

The code here does effectively the same but the magic is hidden in `Wrapf()`.

Reply:
Yes, I changed it like suggested.